### ISSUE TITLE
Support for efficient bulk pdf evaluation for InverseWishart

### ISSUE THREAD

**Issue by bvdmitri** [biaslab/ReactiveMP.jl#187]
Some attempt to optimize bulk in place evaluation of InverseWishart `pdf`

```
function Distributions.pdf!(out::AbstractArray{<:Real}, distribution::ReactiveMP.InverseWishartMessage, samples::AbstractArray{<:AbstractMatrix{<:Real}, O}) where O
    @assert length(out) === length(samples) "Invalid dimensions in pdf!"
    
    p = size(distribution, 1)
    (df, Ψ) = Distributions.params(distribution)
    
    T = copy(Ψ)
    R = similar(T)
    l = length(T)
    
    M = ReactiveMP.fastcholesky!(T)
    
    # logc0 evaluation
    h_df = df / 2
    Ψld = logdet(M)
    logc0 = -h_df * (p * convert(typeof(df), Distributions.logtwo) - Ψld) - StatsFuns.logmvgamma(p, h_df)
    
    # logkernel evaluation 
    @inbounds for i in 1:length(out)
        unsafe_copyto!(T, 1, samples[i], 1, l)
        C = ReactiveMP.fastcholesky!(T)
        ld = logdet(C)
        LinearAlgebra.inv!(C)
        mul!(R, Ψ, C.factors)
        r = tr(R)
        out[i] = exp(-0.5 * ((df + p + 1) * ld + r) + logc0)
    end
    
    return out
end
```

**Comment by albertpod** [comment]
Can I close the issue?

**Comment by bvdmitri** [comment]
@albertpod Yes, has been fixed in #261
